docs/user-guide/src/ironic/ironic-python-agent.md

```diff
@@ -0,0 +1,33 @@
+# ironic python agent (IPA)
+
+IPA is a service written in python that runs within a ramdisk. It provides remote access to  `ironic-conductor` and `ironic-inspector` services to perform various operations on the managed server. It also sends information about the server to `Ironic`.
+
+
+## Data flow
+
+IPA interacts with other components. The information exchanged and the component to which it is sent to or received from are described below.
+The communication between IPA and these component can be encrypted in-transit with SSL/TLS.
+
+- Heartbeat: periodic message informing Ironic that the node is still running.
+- Lookup: data sent to Ironic that helps it determine Ironic’s node UUID for the node.
+- Introspection result: data about hardware details, such as CPU, disk, RAM and Network interfaces.
+
+The above data is is sent/received as follows.
+
+- Lookup/heartbeats data is sent to Ironic.
+- Introspection result is sent to ironic-inspector.
+- user supplied boot image that will be written to the node’s disk is retrieved from HTTPD server
+
+## IPA Configuration
```

Comment:
I'm not sure this section is useful as it is. We need to either mention how it's done in the ironic-image context, or skip it for now.

Reply:
+1 for skipping it.